### PR TITLE
Updating dependencies from last known good builds

### PR DESCRIPTION
**Pull request by eerhardt**
This change is [Reviewable](https://reviewable.io/reviews/dotnet/cli/1990)

### CODE REVIEW

**Comment by eerhardt** [comment]
@piotrpMSFT @anurse @ericstj @livarcocc -

I worked around the breaking changes in CoreFX rc2-23922 with the 2nd commit.  Most notably, I moved ProjectModel and all its dependencies to netstandard1.5.

Please take a look.

File: src/Microsoft.DotNet.TestFramework/project.json [hunk 1]
```diff
@@ -14,7 +14,7 @@
     "NETStandard.Library": "1.5.0-rc2-23922"
   },
   "frameworks": {
-    "netstandard1.3": {
+    "netstandard1.5": {
       "imports": "dnxcore50"
```
Comment (ericstj) [inline src/Microsoft.DotNet.TestFramework/project.json]: It's broken to be importing "dnxcore50" from a "netstandard1.5" library.  That problem needs to be fixed across the repo but not in this change.

**Comment by ericstj** [comment]
Changes look reasonable.  

You'll be able to get rid of those direct references you had to add once your down-stack dependencies upgrade their dependencies on corefx packages.  This started breaking you because we stopped lifting versions to latest in the lineup //cc @davidfowl.

File: src/Microsoft.DotNet.TestFramework/project.json (hunk 1 above)
Comment (anurse) [inline src/Microsoft.DotNet.TestFramework/project.json]: Now that ASP.NET and CoreFX have both transitioned to `netstandard` I think we could probably remove that import.

**Comment by mellinoe** [comment]
We need to update the version of CrossGen used in some of the scripts. The version of CoreCLR used to locate crossgen.exe is hard-coded in CompileTargets.cs. At the very least we need to update it to the version of CoreCLR that is restored by this change. There is a comment in the code saying we should add logic to parse the version of CoreCLR out of the lock file, but we don't necessarily need to go that far with this change.

@weshaggard also just patched up the version of Microsoft.NETCore.App [on the dotnet-core feed](https://dotnet.myget.org/F/dotnet-core/api/v2/package/Microsoft.NETCore.App/1.0.0-rc3-23922), and we need to bring that version over to the cli-deps feed as well.

**Comment by eerhardt** [comment]
@mellinoe - This change updates the version of CrossGen to rc2-23922 https://github.com/dotnet/cli/pull/1990/commits/5818f6e4dfd6c87532cf80dd85513aa497738713#diff-0318e14ec884b5700c20aa3e06d856ceR17.  

Is this what you want?  Or do you want something else?

**Comment by eerhardt** [comment]
Also, we have https://dotnet.myget.org/feed/cli-deps/package/nuget/Microsoft.NETCore.App/1.0.0-rc2-23922 in the cli-deps feed.  Do you need something else?

**Comment by weshaggard** [comment]
@eerhardt I just had a look at https://dotnet.myget.org/feed/cli-deps/package/nuget/Microsoft.NETCore.App/1.0.0-rc2-23922  and it is the unpatched version. It must have been added to clideps before my patching. I will manually update clideps feed as well to get it patched there.

**Comment by weshaggard** [comment]
I just patched it on clideps feed.

**Comment by mellinoe** [comment]
> @mellinoe - This change updates the version of CrossGen to rc2-23922 5818f6e#diff-0318e14ec884b5700c20aa3e06d856ceR17.

> Is this what you want? Or do you want something else?

Sorry, I didn't see that in the diff, that's exactly what I was referring to. :+1: 

Since Wes has patched the clipdeps feed as well, we should be able to remove the CodeAnalysis reference from the sharedframework project.json:

````json
"Microsoft.CodeAnalysis.CSharp": "1.3.0-beta1-20160318-02"
````

That should be included in the NETCore.App package now.

**Comment by mellinoe** [comment]
There's some failures in the CI, and I'm seeing them locally as well:

````
MESSAGE:
Command did not output with Expected Output. Expected: Hello Portable World! File Name: D:\j\workspace\release_windo---4c3b899b\artifacts\win81-x64\stage2\dotnet.exe Arguments: portable Exit Code: 131 StdOut: StdErr: Could not load host policy library []
````
"StdErr: Could not load host policy library []"

I'm not sure how to diagnose that.

**Comment by mellinoe** [comment]
@schellap Could you help diagnose what the problem is here?

**Comment by mellinoe** [comment]
@weshaggard It seems that I'm unable to restore your patched version of Microsoft.NETCore.App from the cli-deps feed. I can download it directly from [here](Eric Mellinocheck out https://dotnet.myget.org/F/cli-deps/api/v2/package/Microsoft.NETCore.App/1.0.0-rc2-23922), but every time I restore it with dotnet, I get the stale version. This is after deleting my nuget cache multiple times.

**Comment by schellap** [comment]
Unless the nupkg contents are unchanged, don't reupload a package to myget. They have some REST level caching going on, so the API will provide the cached version. Cc @brthor

**Comment by weshaggard** [comment]
Generally you can clear your package cache and the http cache for nuget and this works.

At any rate the next build which has this change is now out https://dotnet.myget.org/feed/dotnet-core/package/nuget/Microsoft.NETCore.App/1.0.0-rc2-23923 so we can just jump to that one at this point.

**Comment by eerhardt** [comment]
I've moved up to rc2-23923. @mellinoe - restore was failing because of conflicting versions of Microsoft.CodeAnalysis.CSharp

http://dotnet-ci.cloudapp.net/job/dotnet_cli/job/rel_1.0.0/job/release_ubuntu_x64_prtest/368/console

```
07:14:18 Errors in /opt/code/src/sharedframework/framework/project.json
07:14:18     Version conflict detected for Microsoft.CodeAnalysis.Common.
07:14:18      framework (>= 1.0.0) -> Microsoft.NETCore.App (>= 1.0.0-rc2-23923) -> Microsoft.CodeAnalysis.VisualBasic (>= 1.3.0-beta1-20160321-04) -> Microsoft.CodeAnalysis.Common (>= 1.3.0-beta1-20160321-04)
07:14:18      framework (>= 1.0.0) -> Microsoft.CodeAnalysis.CSharp (>= 1.3.0-beta1-20160318-02) -> Microsoft.CodeAnalysis.Common (= 1.3.0-beta1-20160318-02).
```

So I removed the CodeAnalysis reference from the sharedframework project.json.

**Comment by eerhardt** [comment]
I think I've found the issue with `Could not load host policy library []`.  Luckily moving from build 23922 to 23923 on my machine allowed the bug to rear its evil head on my local machine.

The issue is we have a new 'TestAssets\TestPackage' called 'dotnet-portable', which is used as a packaged command in our tests. This package is always created as version "1.0.0", even though it changes as we update the project.json files. If a machine had an old version of this package cached in its NuGet cache, it was always picking up the old version. The NuGet package references which shared framework build it uses - rc2-23911, rc2-23922, etc.  When you move up to a newer CoreFX build, these old shared framework folders are no longer there, but the NuGet package references the old version. Thus it can't find the hostpolicy.dll, and it fails.

Fixing this by cleaning the dotnet-portable test nuget packages from the cache.  @brthor can you take a look at my latest commit?

@schellap - Can we get a better error message in this case? It would have really helped trace down the issue if the directory it was looking in would have been printed out instead of just `Could not load host policy library []`

**Comment by schellap** [comment]
> If a machine had an old version of this package cached in its NuGet cache, it was always picking up the old version.

The CI mahine clears its cache for each run, right?

> Can we get a better error message in this case?
I will get a fix for that.
https://github.com/dotnet/cli/issues/2008

**Comment by eerhardt** [comment]
> The CI mahine clears its cache for each run, right?

No, not necessarily. We keep the NuGet cache across runs in order to speed up the build, and take load off the server.  (Supposedly this is OK because packages *should* be changing versions when the content changes. But that isn't the case with these test packages.)

**Comment by mellinoe** [comment]
> So I removed the CodeAnalysis reference from the sharedframework project.json.

That sounds correct to me. The reference was only there to force an upgrade from the version in NETCore.App. Since NETCore.App has a higher version, deleting it is good.

Thanks for investigating the issue, this was a bit of a frustrating one to figure out :anguished: .

File: scripts/dotnet-cli-build/TestTargets.cs [hunk 2]
```diff
@@ -120,6 +120,8 @@ public static BuildTargetResult BuildTestAssetPackages(BuildTargetContext c)
         public static BuildTargetResult CleanTestPackages(BuildTargetContext c)
         {
             Rmdir(Path.Combine(Dirs.NuGetPackages, "dotnet-hello"));
+            Rmdir(Path.Combine(Dirs.NuGetPackages, "dotnet-portable"));
```
Comment (brthor) [inline scripts/dotnet-cli-build/TestTargets.cs]: :+1:

**Comment by brthor** [comment]
good catch on dotnet-portable @eerhardt  LGTM

**Comment by mellinoe** [comment]
It seems like the non-Windows runs are hanging on `dotnet-projectmodel-server.Tests`. I'll see if I can repro locally.

**Comment by mellinoe** [comment]
I'm able to reproduce it on my Mac. This line [here](https://github.com/dotnet/cli/blob/rel/1.0.0/test/dotnet-projectmodel-server.Tests/DthTests.cs#L90) is hanging indefinitely for me. It ultimately calls through to [BinaryWriter.Write](https://github.com/dotnet/cli/blob/rel/1.0.0/test/dotnet-projectmodel-server.Tests/DthTestClient.cs#L92) where the BinaryWriter is writing to a NetworkStream. That is the call that hangs indefinitely in my repro. Perhaps there's a regression in the latest NetworkStream?

@eerhardt Have you been able to reproduce this?

**Comment by eerhardt** [comment]
> @eerhardt Have you been able to reproduce this?

Yes, I looped in @ericeil to help investigate.  It looks like there has been some changes in the Socket Write area in the last week: https://github.com/dotnet/corefx/pull/6838

/cc @stephentoub in case he has any ideas why writing to a Socket is hanging on Unix, and how to diagnose the problem.

**Comment by ericeil** [comment]
> I looped in @ericeil to help investigate.

I have a theory as to what is happening; putting together a fix to try.

**Comment by ericeil** [comment]
This is definitely a regression in Socket.  I've submitted a [CoreFx PR](https://github.com/dotnet/corefx/pull/7184) to fix it.

**Comment by piotrpMSFT** [comment]
one down!!!

**Comment by anurse** [comment]
7 to go, aka 3 hours or so :trollface:

**Comment by piotrpMSFT** [comment]
NOOOOOO

**Comment by anurse** [comment]
I can take this over to a branch and fix it. I think it's related to rebasing on my changes (weird that the other OSes passed...)

**Comment by anurse** [comment]
I can't repro the issue on Windows... that's really strange.

**Comment by livarcocc** [comment]
@ellismg @stephentoub They helped me deal with a similar error a few weeks ago.

**Comment by anurse** [comment]
```
14:42:05 
/mnt/resource/j/workspace/dotnet_cli/rel_1.0.0/debug_centos7.1_x64_prtest/src/Microsoft.Extensions.DependencyModel/src/Microsoft.Extensions.DependencyModel/DependencyContextLoader.cs(47,51): error CS0117: 'Assembly' does not contain a definition for 'GetEntryAssembly'
14:42:05 /mnt/resource/j/workspace/dotnet_cli/rel_1.0.0/debug_centos7.1_x64_prtest/src/Microsoft.Extensions.DependencyModel/src/Microsoft.Extensions.DependencyModel/DependencyContextLoader.cs(121,62): error CS1061: 'Assembly' does not contain a definition for 'Location' and no extension method 'Location' accepting a first argument of type 'Assembly' could be found (are you missing a using directive or an assembly reference?)
14:42:05 /mnt/resource/j/workspace/dotnet_cli/rel_1.0.0/debug_centos7.1_x64_prtest/src/Microsoft.Extensions.DependencyModel/src/Microsoft.Extensions.DependencyModel/DependencyContext.cs(109,66): error CS0117: 'Assembly' does not contain a definition for 'GetEntryAssembly'
14:42:05 
```

Seems to only repro on CentOS and Ubuntu, which is weird because these should be in the `ref`s, which aren't OS-dependent.

**Comment by ellismg** [comment]
Can you extract the contents of the lock file and see what it would be using to compile against?

**Comment by anurse** [comment]
Ok, this is really weird. Windows lock files are building against the `netstandard1.5`, but Cent and Ubuntu are using the `netstandard1.3` one. The members were **added** in `1.5`, so it makes sense that that is why Cent/Ubuntu are failing. What's really weird is that the project does **target** `netstandard1.3`, so I'm not sure why the other builds are getting `1.5`...

**Comment by anurse** [comment]
I've moved DependencyModel to 1.5 which should fix this (since we're OK bumping this). Still kinda nervous about how the versions got split like that...

I've opened a new PR for my branch (which includes these commits plus the fix to move DependencyModel forward): https://github.com/dotnet/cli/pull/2040

**Comment by anurse** [comment]
Discussion can continue over there, or here. I don't mind :)

**Pull request closed without merging**